src/NServiceBus.Core/Impersonation/Windows/WindowsIdentityEnricher.cs

```diff
@@ -19,11 +18,7 @@ public void MutateOutgoing(LogicalMessage logicalMessage, TransportMessage trans
                 return;
             }
             var windowsIdentity = WindowsIdentity.GetCurrent();
-            if (windowsIdentity != null)
```

Comment:
@johnsimons look closer. https://referencesource.microsoft.com/#mscorlib/system/security/principal/windowsidentity.cs,807 threadonly is always false. http://stackoverflow.com/questions/15998381/can-windowsidentity-getcurrent-return-null